### PR TITLE
Add Angular "Engineering Notes"

### LINKED ISSUE #1689: Incorporate or link to dart-lang/angular/doc
The dart-lang/angular repo's [doc directory](https://github.com/dart-lang/angular/tree/master/doc) has a bunch of good docs that need to be incorporated into or referred to by the webdev.dartlang.org docs. I'll update the deferred loading reference right now, but we need to do much more.

Related: dart-lang/angular#1452.

---

- [ ] separate the notes by subdirectory, and have descriptions of what (generally) is in each subdirectory
- [ ] link to the source directory (or directories)
- [ ] Exclude internally-focused `developing/` directory

### PR DESCRIPTION
**Pull request by chalin**
A first cut at publishing angular/doc "engineering notes" under webdev. 

Staged at https://angulardart-org-dev.firebaseapp.com/angular/note, which is an index page listing all notes in alphabetical order.

Notes:
- Jekyll pages are **automatically generated** (using `gulp note-refresh`) from the angular/doc folder. 
  - **Only pages with a heading at level 1** are processed (since that heading becomes the page title).
- Note that the GH markdown used in the original pages doesn't match the default Kramdown syntax (so pages are slightly reformatted).

Fixes #1689

### CODE REVIEW

File: src/angular/note/index.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+toc: false
+---
+
+{{page.description}}
+
+{% assign pages = site.pages | sort: 'title' %}
+{% for note in pages -%}
+  {% assign pageUrl = note.url | regex_replace: '^/angular/note', '!!!' -%}
+  {% unless pageUrl contains '!!!' %}{% continue %}{% endunless -%}
+  - [{{ note.title }}]({{ note.url }})
```
Comment (kwalrath) [inline src/angular/note/index.md]: Overview: Angular Engineer Notes appears in the list. :)

**Comment by kwalrath** [comment]
Cool!

It seems like we should separate the notes by subdirectory, and have descriptions of what (generally) is in each subdirectory.

In addition, we should link to the source directory (or directories) but probably shouldn't include stuff under the internally-focused developing/ directory.

File: src/angular/note/index.md (hunk 1 above)
Comment (chalin) [reply src/angular/note/index.md]: Oops, I had that removed in the previous version of the page.
Fixed.

**Comment by kwalrath** [comment]
How do you think we should surface these notes in the navigation?

**Comment by chalin** [comment]
> How do you think we should surface these notes in the navigation?

In the sidenav, something like this:
> ANGULAR DART
> Guide >
> Tutorial >
> Advanced >
> Engineering Notes

Or possibly just "Notes". WDYT?

**Comment by chalin** [comment]
> It seems like we should separate the notes by subdirectory, and have descriptions of what (generally) is in each subdirectory.
>
> In addition, we should link to the source directory (or directories) but probably shouldn't include stuff under the internally-focused developing/ directory.

Ok, I'll make those changes.

**Comment by kwalrath** [comment]
> > How do you think we should surface these notes in the navigation?
> 
> In the sidenav, something like this:
> > ANGULAR DART
> > Guide >
> > Tutorial >
> > Advanced >
> > Engineering Notes
> 
> Or possibly just "Notes". WDYT?

I like "Notes". (The other one isn't bad, either, but short is good.)

**Comment by kwalrath** [comment]
This page should link to other notes, too — release notes, e.g. That could be in some pre-fab text at the top (as opposed to being auto-generated).

**Comment by chalin** [comment]
@kwalrath - I've organized the notes by topic rather than matching the directory structure exactly. Also, I've followed the Tutorials page formatting, which looks like this:

screen shot 2018-07-11 at 07 50 56

This is what the **Notes** page looks like now:

screen shot 2018-07-11 at 07 57 02


WDYT?

**Comment by chalin** [comment]
@kwalrath: the note page has been reorganized, and a link was added to the sidenav. I suggest not linking to the the note sources in this PR. Staging server refreshed. PTAL

**Comment by kwalrath** [comment]
Although I like the categories, sorting by title rather than subdirectory seems fragile. What do you think, @alorenzen & @matanlurey?

**Comment by kwalrath** [comment]
Build failed due to bad internal links in the notes:

```
http://localhost:4001/angular/note/effective/change-detection
- (367:10) 'PREFER s..' => http://localhost:4001/angular/note/effective/change-detection#prefer-settings-to-ngafterxchecked (HTTP 200 but missing anchor)
http://localhost:4001/angular/note/effective/di
- (568:21) 'DO use t..' => http://localhost:4001/angular/note/effective/di#... (HTTP 200 but missing anchor)
- (568:117) 'DO use t..' => http://localhost:4001/angular/note/effective/di#... (HTTP 200 but missing anchor)
http://localhost:4001/angular/note/faq/component-loading
- (379:7) 'migration' => http://localhost:4001/angular/note/faq/component-loading#Migration (HTTP 200 but missing anchor)
http://localhost:4001/angular/note/faq/di
- (552:42) 'dependen..' => http://localhost:4001/angular/note/effective/di.md#components (HTTP 404)
```

**Comment by chalin** [comment]
> Although I like the categories, sorting by title rather than subdirectory seems fragile.

Think of it as classifying notes by tag. It's just that we don't have any Jekyll frontmatter in the original pages, so I'm fetching "tags" from the title. IMHO this works well enough for now. We can always reorganize later as needed.

**Comment by alorenzen** [comment]
I know it's not written as such, but `Imperative Component Loading` is actually listed as one of our FAQs internally. Instead of listing it under "Other", do you have a separate recommendation for what we could call it? If so, we can make that change in our repo as well.

**Comment by chalin** [comment]
> I know it's not written as such, but Imperative Component Loading is actually listed as one of our FAQs internally. Instead of listing it under "Other", do you have a separate recommendation for what we could call it? If so, we can make that change in our repo as well.

If you'd prefer that I present the notes grouped in a manner that matches the subdirectories under `angular/doc` I can do that. (In fact, that was Kathy's original suggestion.)

> As a whole, I'm happy with this solution for publishing our docs on webdev site. As we go along, we can improve the process, but for now I think this is a good first step.

Great!

> Other than the overall navigation, I didn't really review each of the individual pages. 

That's fine. The pages are essentially the same as those in the angular monorepo.

**Comment by chalin** [comment]
There is a Notes index page section for each directory under angular/doc, except for:

- developing, which isn't shown because it is internal
- router, which contains only one migration guide, and that guide is shown in the Migration section

The staging server has been refreshed, PTAL.

**Pull request merged**